### PR TITLE
exlcude defaults from validation

### LINKED ISSUE #10247: Unable to finetune on rc2
### Rasa Open Source version

3.0.0rc2

### Rasa SDK version

_No response_

### Rasa X version

_No response_

### Python version

3.8

### What operating system are you using?

OSX

### What happened?

Trained an initial model with `rasa init`, added a new training example to one of the existing intents and ran `rasa train --finetune`. Fails with:

```
InvalidConfigException: Cannot finetune because more than just the 'epoch' keys have been changed in the configuration. Please revert your configuration and only change the 'epoch' settings where needed.
```

Might be related to https://github.com/RasaHQ/rasa/issues/10246

### Command / Request

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by ka-bu**
**Proposed changes**:
- closes #10247 and #10246

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by carlad** [comment]
Do we also want to drop the relevant CLI defaults when training?

**Review by carlad — APPROVED** [review]

File: rasa/cli/arguments/train.py [hunk 1]
```diff
@@ -151,7 +153,7 @@ def add_augmentation_param(
     parser.add_argument(
         "--augmentation",
         type=int,
-        default=50,
+        default=TrainingTrackerProvider.get_default_config()['augmentation_factor'],
```
Comment (usc-m) [inline rasa/cli/arguments/train.py:156]: Oh this isn't actually as bad as I thought now that I see it written out - actually happier with this now. Still think it's a bit uncomfortable but it's not the worst

File: rasa/engine/graph.py [hunk 2]
```diff
@@ -22,40 +22,44 @@
...
+        uses: The class which models the behavior of this specific graph node.
+        constructor_name: The name of the constructor which should be used to
+            instantiate the component. If `eager==False` then the `constructor` can
+            also specify parameters which are filled by parent nodes. This is e.g.
+            useful if a parent node returns a `Resource` and this node wants to
+            directly load itself from this resource.
+        fn: The name of the function which should be called on the instantiated
+            component when the graph is executed. The parameters from `needs` are
+            filled from the parent nodes.
+        config: The configuration for this graph node.
```
Comment (usc-m) [inline rasa/engine/graph.py:39]: I wonder if we shouldn't change this to mention that this is the config specified by the user, which will then be merged with the default config?

File: rasa/cli/arguments/train.py (hunk 1 above)
Comment (ka-bu) [reply rasa/cli/arguments/train.py:156]: 🥳

**Review by usc-m — APPROVED** [review]
🎉

File: rasa/engine/graph.py [hunk 3]
```diff
@@ -36,7 +36,9 @@ class SchemaNode:
         fn: The name of the function which should be called on the instantiated
             component when the graph is executed. The parameters from `needs` are
             filled from the parent nodes.
-        config: The configuration for this graph node.
+        config: The user's configuration for this graph node. This configuration
+            does not need to be specify all possible parameters; the default values
```
Comment (carlad) [inline rasa/engine/graph.py:40]: A small typo here: `does not need to be specify` should be `does not need to specify`

**Pull request merged**